src/sentry/api/endpoints/event_grouping_info.py

```diff
@@ -24,7 +24,8 @@ def get(self, request, project, event_id):
         if event is None:
             raise ResourceDoesNotExist
 
-        event.bind_node_data()
+        if not options.get("eventstore.use-nodestore"):
+            event.bind_node_data()
```

Comment:
Why can't you make bind_node_data a nop if the data is already bound? Having the callsite of eventstore and event knowing whether they need to bind data or not depending on where the data was loaded from seems very tight coupling with the eventstore implementation. This should be transparent.

Reply:
Well, I've recently realised it's going to be far more efficient to fetch both single events (tackled in this PR) as well as collections of events (https://github.com/getsentry/sentry/pull/16230) from nodestore instead of snuba. As a result, eventstore will never need to return an unbound event again. Since no one will ever need to manually bind nodes again, we can make binding private to eventstore and remove it everywhere else. No one else would need to worry about binding or having incomplete events again, it won't be a decision to for anyone to make. I put the binding behind the switch for the rollout to prevent double fetching, but when we verify it works it can be removed.

Reply:
Ok, then let's make bind_node a nop and remove all the logic altogether when you remove the option to rollout. What happens if someone introduces new behavior without binding nodes seeing how it works and then we revert ?

Reply:
That seems ok. I can't think of any scenario we ever need to force reload from nodestore or manually bind some other data to an event, can you?

Reply:
The SnubaEvent/NodeData model has a mechanism where it automatically loads its node data on any attempt to access unfetched data, so I don't think anyone introducing new behavior has a chance of breaking things even if we flip the switch and they forget to bind.

Reply:
I am ok if you just remove bind_nodes for good immediately.

Reply:
Actually, we need to hang on to bind_nodes for RawEvent and Django event right now. But we won't use it for regular Events.

Reply:
Then it seems the condition on whether bind_nodes does something or not depends on the class and not on the option. Can we always call bind and have this condition in the method itself?
My concern is that the callsite that uses the event should not bother of how data is bound on different types of events. Either they always bind or they never bind.